python/cugraph/centrality/katz_centrality.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (c) 2019, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from cugraph.centrality import katz_centrality_wrapper
+
+
+def katz_centrality(G,
+                    alpha=0.1,
+                    max_iter=100,
+                    tol=1.0e-5,
+                    nstart=None,
+                    normalized=True):
+    """
+    Compute the Katz centrality for the nodes of the graph G
+
+    Parameters
+    ----------
+    graph : cuGraph.Graph
+        cuGraph graph descriptor with connectivity information. Edge weights,
+        if present, should be single or double precision floating point values.
+    alpha : float
+        Attenuation factor with a default value of 0.1. Alpha is set to
+        1/(lambda_max) if it is greater where lambda_max is the maximum degree
+        of the graph.
+    max_iter : int
+        The maximum number of iterations before an answer is returned. This can
+        be used to limit the execution time and do an early exit before the
+        solver reaches the convergence tolerance.
+        If this value is lower or equal to 0 cuGraph will use the default
+        value, which is 100.
```

Comment:
Nx uses 1000. Not sure it is a good idea though. What do you think?

Reply:
We can definitely use 1000. NetworkX uses a different convergence technique than what we use. So this parameter is slightly different. 1000 will increase the execution by 10X without seeing a big change in performance. I won't be surprised if 10 or 20 iterations are enough.

Reply:
I see. Does it worth mentioning that we use a different convergence technique in the doc then? Or we could just say something like "For most cases, the ranking does not change much after 20 interations with our implementation"

Reply:
Good question. Maybe we can run some experiments to figure out what a good iteration count is for the average case.